src/common/pool_map.c

```diff
@@ -1812,6 +1812,54 @@ pool_map_find_failed_tgts(struct pool_map *map, struct pool_target **tgt_pp,
 				  tgt_cnt);
 }
 
+
+/**
+ * Find all targets with @status in specific rank. Note: &tgt_pp will be
+ * allocated if it is NULL, then the caller is reponsible to free it.
```

Comment:
Indeed. But my  thought would be the caller should make sure to provide enough buffer here, since there would be limited number of the targets per server anyway. 
Probably pass in NULL would be easier anyway.